standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java

```diff
@@ -546,7 +552,15 @@ public static void updateTableStatsForCreateTable(Warehouse wh, Database db, Tab
       }
     }
 
-    if (MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.STATS_AUTO_GATHER)) {
+    IcebergStatsSource statsSource = IcebergStatsSource.fromString(
```

Comment:
could we move it into method and evaluate only if table is iceberg?

Reply:
done